src/tonic/api/points_api.rs

```diff
@@ -60,6 +72,168 @@ impl Points for PointsService {
         let response = PointsOperationResponse::from((timing, result));
         Ok(Response::new(response))
     }
+
+    async fn delete(
+        &self,
+        request: Request<DeletePoints>,
+    ) -> Result<Response<PointsOperationResponse>, Status> {
+        let DeletePoints {
+            collection,
+            wait,
+            points,
+        } = request.into_inner();
+
+        let points_selector = points.expect("missing PointInsertOperation").try_into()?;
+
+        let timing = Instant::now();
+        let result = do_delete_points(
+            self.toc.as_ref(),
+            &collection,
+            points_selector,
+            wait.unwrap_or(false),
+        )
+        .await
+        .map_err(error_to_status)?;
+
+        let response = PointsOperationResponse::from((timing, result));
+        Ok(Response::new(response))
+    }
+
+    async fn set_payload(
+        &self,
+        request: Request<SetPayloadPoints>,
+    ) -> Result<Response<PointsOperationResponse>, Status> {
+        let SetPayloadPoints {
+            collection,
+            wait,
+            payload,
+            points,
+        } = request.into_inner();
+
+        let operation = collection::operations::payload_ops::SetPayload {
+            payload: payload_to_interface(payload)?,
+            points: points.into_iter().map(|p| p.into()).collect(),
+        };
+
+        let timing = Instant::now();
+        let result = do_set_payload(
+            self.toc.as_ref(),
+            &collection,
+            operation,
+            wait.unwrap_or(false),
+        )
+        .await
+        .map_err(error_to_status)?;
+
+        let response = PointsOperationResponse::from((timing, result));
+        Ok(Response::new(response))
+    }
+
+    async fn delete_payload(
+        &self,
+        request: Request<DeletePayloadPoints>,
+    ) -> Result<Response<PointsOperationResponse>, Status> {
+        let DeletePayloadPoints {
+            collection,
+            wait,
+            keys,
+            points,
+        } = request.into_inner();
+
+        let operation = DeletePayload {
+            keys,
+            points: points.into_iter().map(|p| p.into()).collect(),
+        };
+
+        let timing = Instant::now();
+        let result = do_delete_payload(
+            self.toc.as_ref(),
+            &collection,
+            operation,
+            wait.unwrap_or(false),
+        )
+        .await
+        .map_err(error_to_status)?;
+
+        let response = PointsOperationResponse::from((timing, result));
+        Ok(Response::new(response))
+    }
+
+    async fn clear_payload(
+        &self,
+        request: Request<ClearPayloadPoints>,
+    ) -> Result<Response<PointsOperationResponse>, Status> {
+        let ClearPayloadPoints {
+            collection,
+            wait,
+            points,
+        } = request.into_inner();
+
+        let points_selector = points.expect("Missing points selector").try_into()?;
```

Comment:
`expect("...")` will cause panic, won't it?

Reply:
Good catch :+1: 
https://github.com/qdrant/qdrant/pull/264/commits/d83a07172b3bcf4ddf30404815c5e65962f9750c